core/src/main/java/hudson/security/GlobalSecurityConfiguration.java

```diff
@@ -111,14 +111,10 @@ public boolean configure(StaplerRequest req, JSONObject json) throws hudson.mode
         // for compatibility reasons, the actual value is stored in Jenkins
         Jenkins j = Jenkins.getInstance();
         j.checkPermission(Jenkins.ADMINISTER);
-        if (json.has("useSecurity")) {
-            JSONObject security = json.getJSONObject("useSecurity");
-            j.setDisableRememberMe(security.optBoolean("disableRememberMe", false));
-            j.setSecurityRealm(SecurityRealm.all().newInstanceFromRadioList(security, "realm"));
-            j.setAuthorizationStrategy(AuthorizationStrategy.all().newInstanceFromRadioList(security, "authorization"));    
-        } else {
-            j.disableSecurity();
-        }
+
+        j.setDisableRememberMe(json.optBoolean("disableRememberMe", false));
+        j.setSecurityRealm(SecurityRealm.all().newInstanceFromRadioList(json, "realm"));
```

Comment:
Validation of radio-selectors is PITA though doable.

I agree that additional check would be useful. Maybe `Jenkins.getAuthorizationStrategy()` should just ignore the setting when the security realm is not set. And them we could hook on the existing Security Administrative Monitor